pennylane/devices/qubit/simulate.py

```diff
@@ -65,6 +65,40 @@
 }
 
 
+class TreeTraversalStack:
+    """This class is used to record various data used during the
+    depth-first tree-traversal procedure for simulating dynamic circuits."""
+
+    counts: Sequence
+    probs: Sequence
+    results_0: Sequence
+    results_1: Sequence
+    states: Sequence
+
+    def __init__(self, max_depth):
+        self.counts = [None] * max_depth
```

Comment:
Might be simple questions, but:

1) What does `None` indicate?
2) What will be the values populated into the list?

Reply:
1. It's a placeholder.
2. The counts, probs or states at a given depth/node (and all parent nodes). Similarly for results, either it stores the terminal measurements of a given leaf if the depth is maximal or the aggregate measurements of a subtree if the depth is intermediate. As soon as aggregate measurements are present in both branches, they are merged and the depth is decremented.